File: typescript/packages/bench/src/rules/partAvailabilityRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class PartAvailabilityRule implements RuleContract {
    readonly key = "part-availability";

    evaluate(data: Dataset): boolean {
        return data.parts.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/part_availability_rule.rs
//! Rule 6/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct PartAvailabilityRule;

impl RuleContract for PartAvailabilityRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.parts.is_empty()
    }
}
